src/stactools/sentinel2/stac.py

```diff
@@ -311,7 +315,25 @@ def set_asset_properties(_asset: pystac.Asset, _band_gsd: Optional[int] = None):
         asset_eo = EOExtension.ext(asset)
         asset_eo.bands = [band_from_band_id(band_id)]
         set_asset_properties(asset, band_gsd)
-        return asset_key, asset
+
+        if processing_baseline < "04.00":
+            scale = 0.0001  # DN = 10000 * REFLECTANCE
+            offset = 0
+        else:  # todo: pull this from metadata
+            scale = 0.0001
+            offset = -1000
+
+        RasterExtension.ext(asset).bands = [
+            RasterBand.create(
+                nodata=0,
+                spatial_resolution=resolution,
+                data_type=DataType.UINT16,
+                bits_per_sample=15,
+                unit="unitless",
```

Comment:
In the planetary computer, we've been omitting the unit if there isn't one. I don't think it's wrong to leave it as "unitless", but we probably should put some sort of guidance in the raster extension about the correct approach.

Reply:
I think the right thing to do here is to use `unitless` to clarify the difference between the unit being unitless, and the unit not being known or just not being populated.

Reply:
I'm going to create a raster extension PR to clarify this and fix the incorrect S2 example.